crates/bevy_ecs/macros/src/world_query_data.rs

```diff
@@ -29,7 +29,7 @@ mod field_attr_keywords {
     syn::custom_keyword!(ignore);
 }
 
-pub static WORLD_QUERY_DATA_ATTRIBUTE_NAME: &str = "world_query_data";
+pub static WORLD_QUERY_DATA_ATTRIBUTE_NAME: &str = "query_data";
```

Comment:
Perhaps remove `WORLD` here as well?

Reply:
I was contemplating this as well. How about `derive_world_query_filter`, `derive_world_query_filter_impl` fns and their helper structs? Does it make sense to also rename the source file as well?

Reply:
Imo it doesn't make sense to leave traces of the old naming that might mislead future contributors, but in the end these internal names/filenames aren't going to affect users so it's not a huge deal.

Reply:
Gotcha, I'm leaning on cleaning it up completely. Do you mind if I re-request for a review after the changes?